### PR TITLE
Bootstrap pip installation

### LINKED ISSUE #243: Unable to install pip using pip provider
Pull request #241 made it possible to install `pip` via `pip` with the following manifest:

```
class { 'python':
  ensure => 'present',
  provider  => 'pip',
}
```

This does not quite work as expected, however, because it presents a bootstrapping issue (it is not possible to install `pip` via `pip` if `pip` is not installed).

I'm not sure what the best way to fix this is...

### PR DESCRIPTION
**Pull request by joshuaspence**
Fixes #243.

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -65,6 +65,14 @@
 
   case $python::provider {
     pip: {
+      # Install pip without pip, see https://pip.pypa.io/en/stable/installing/.
+      exec { 'bootstrap pip':
+        command => 'curl https://bootstrap.pypa.io/get-pip.py | python',
+        creates => '/usr/local/bin/pip',
```
Comment (joshuaspence) [inline manifests/install.pp]: This is probably not platform independent.

Comment (shivapoudel) [inline manifests/install.pp]: @joshuaspence ^^

Comment (joshuaspence) [inline manifests/install.pp]: ?

Comment (shivapoudel) [inline manifests/install.pp]: If you do it @joshuaspence :-)

File: manifests/install.pp [hunk 2]
```diff
@@ -65,6 +65,14 @@
 
   case $python::provider {
     pip: {
+      # Install pip without pip, see https://pip.pypa.io/en/stable/installing/.
+      exec { 'bootstrap pip':
+        command => 'curl https://bootstrap.pypa.io/get-pip.py | python',
```
Comment (ghoneycutt) [inline manifests/install.pp]: Downloading scripts from the internet and running them makes baby jesus cry. Can't we just install the packages?

File: manifests/install.pp (hunk 1 above)
Comment (joshuaspence) [inline manifests/install.pp]: Ok

File: manifests/install.pp (hunk 2 above)
Comment (joshuaspence) [inline manifests/install.pp]: This is only done if you choose `pip` as a provider.

Generally I'd agree with you, but I couldn't think of a better way to do this whilst remaining cross-platform compatible.

**Pull request merged**